src/dmtcp_dlsym.cpp

```diff
@@ -612,13 +612,13 @@ dmtcp_dlsym_lib(const char *libname, const char *symbol)
 
 /*
  * Returns the offset of the given function within the given shared library
- * or -1 if the function does not exist in the library
+ * or LIB_FNC_OFFSET_FAILED if the function does not exist in the library
  */
-EXTERNC ptrdiff_t
+EXTERNC uint64_t
 dmtcp_dlsym_lib_fnc_offset(const char *libname, const char *symbol)
 {
   dt_tag tags;
-  ptrdiff_t ret = -1;
+  uint64_t ret = LIB_FNC_OFFSET_FAILED;
```

Comment:
Lower down, just after line 631, I should add `JASSERT(ret >= 0)(result)(tags.base_addr)`.

I should then add a comment that `ret` is either positive (non-negative), or else -1 (LIB_FNC_OFFSET_FAILED).  This is in analogy with 'mmap' returning either an unsigned 'void *' or -1 (MAP_FAILED).

Reply:
>     1. In the "Files changed", I've now added an in-line comment proposing that I add a `JASSERT(ret >= 0);`, and then some further comments of clarification.  This might explain it to others who read this code.

Now I'm more confused :-) Doesn't casting `ret` to `uint64_t` make it always non-negative?